sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_requiredparameters.json

```diff
@@ -2108,7 +2110,8 @@
                         "type": "string"
                     }
                 }
-            }
+            },
+            "required": ["id"]
```

Comment:
Are you sure this is required and not part of the APIs? In other words, will we get a 400 if we do not set this?

Reply:
So this is moreso about if the field is required when building the defined type here, and not so much to do with how that type gets used in the service APIs. The only real impact of this change should be that our autogenerated classes will require this id field be specified at constructor time

Reply:
```csharp
public Device();
```

vs

```csharp
public Device(string id);
```